src/server.py

```diff
@@ -333,6 +333,10 @@ class HackSoundServer(Gio.Application):
         <method name='StopSound'>
           <arg type='s' name='uuid' direction='in'/>
         </method>
+        <method name='StopSoundAtBusName'>
+          <arg type='s' name='uuid' direction='in'/>
+          <arg type='s' name='bus_name' direction='in'/>
+        </method>
```

Comment:
(in case i missed some conversation that happened outside phabricator/github).

Why "limiting" this feature to be "per-bus-name"? Is there a reason for not doing this a global thing? See example below:

Say its called `PlaySoundGlobal('id')` and it would ensure that only one instance is played at once, for all the apps globally, e.g. 

```
PlaySoundGlobal('hack-toolbox/lockscreen/inactive`);
StopSoundGlobal('hack-toolbox/lockscreen/inactive');
```

This would not only simplify things, specially on the toolbox side, but it would also address the problem of having 2-3 toolboxes opened and spamming with the active sound.

It would still require the refcount (or similar) logic to ensure that the sound is played while all the sender haven't stopped the sound (or quit), but it should be simpler.

Reply:
The problem with this is to suppose that by adding a StopSound call in your code it will always be called. If the application crashes (or is killed) if it is a looping sound it will kept playing forever. 

Then if I have a PlaySoundGlobal method and the caller application dies, the refcount for this sound may never reach 0.

More feedback is still appreciated.

Reply:
If I use a PlaySoundGlobal method how would I know when to stop the sound because an application died?

Reply:
I thought you were already detecting if the sender app died/closed, isn't the case?

But, If that is the case, then you `refcount--;` when the app dies/closes, and then check stop it when `refcount==0`.

Reply:
Oh, maybe the toolbox never closes? That would be a problem.

Reply:
> I thought you were already detecting if the sender app died/closed, isn't the case?

Yes, I detect if the app dies/closes.

> But, If that is the case, then you refcount--; when the app dies/closes, and then check stop it when refcount==0.

Yes, but if I don't track the refcount per bus name as I am doing in this commit, how can I know which sound reference count to decrease because of the closed application?

Reply:
if it is a global sound, there is only 1 global refcount. You keep track of clients that "play" it, you decreased the global refcount for that global instance when each of these clients globalStops/dies/closes. Any reason for that not to work? (honestly question)